### ISSUE TITLE
remove both utils.py?

### ISSUE THREAD

**Issue by jeff-regier** [YosefLab/scVI#101]
@Edouard360  Could we move the functions in `scvi/utils.py` and `scvi/models/utils.py` closer to where they are used, and remove those files? Files full of helper functions work best when those utilities are used all over the place. But most these functions seemed to be used just once. It might be easier on someone reading the code if we put them next to where they are used and delete the `utils.py` files.

Also, for the `to_cuda()` function, I wonder if we might be able to include it somehow in the `DataLoader` class, so that the data loader automatically returns data that is on the gpu if a boolean `use_cuda` variable is set:
https://github.com/YosefLab/scVI/blob/02d7dffa9c773f80b35c6fa62080d91116e269df/scvi/utils.py#L26-L27

**Comment by jeff-regier** [comment]
OK I see `no_grad` is used a lot of places, but wasn't there some setting so that torch doesn't compute gradients by default? It's just the training loop where gradients are necessary. Everywhere else we're not computing gradients, and we're in evaluation mode. (?)

**Comment by Edouard360** [comment]
What do you think of introducing:

https://github.com/YosefLab/scVI/blob/b73ba23cb341aee3952d38c02a64ae092df74123/scvi/dataset/data_loaders.py#L10-L20

**Comment by jeff-regier** [comment]
Sure, looks good!

**Comment by Edouard360** [comment]
The `broadcast_labels` function however is needed for both semi supervised model (VAEC, SVAEC) ... I could probably factorize it as a static function, and put it under a new Semi Supervised model class or under VAEC if we choose SVAEC to be child class of SVAEC.

**Comment by Edouard360** [comment]
As for the gradients, I think we still need to put `to.no_grad()` in most places.

torch.set_grad_enabled(False) doesn't seem to work if only called at one place

**Comment by jeff-regier** [comment]
Hmm..I see. I thought `torch.set_grad_enabled(False)` would let us change the default to no_grad. Maybe those `@no_grad` decorators are a good way to go.

**Comment by jeff-regier** [comment]
Does it matter where you call `torch.set_grad_enabled(False)`? Maybe right after the import statements? (but outside of any class)

**Comment by Edouard360** [comment]
I had precisely tried outside of the class - though not just after import statements, I will try

**Comment by Edouard360** [comment]
Ok I changed it in b162b4a in #112.

It's in Inference:

https://github.com/YosefLab/scVI/blob/b162b4a0132432ba86531b18099a89e15b32fbba/scvi/inference/inference.py#L9

Did you rather see it somewhere else ?

**Comment by jeff-regier** [comment]
That seems good -- thanks for making the change.
